### ISSUE TITLE
OMX_ErrorInsufficientResources

### ISSUE THREAD

**Issue by RidaAyed** [tizonia/tizonia-openmax-il#501]
Hi

Does anyone see what could be wrong?
```
master└─ $ tizonia --spotify-album thriller
tizonia 0.15.0. Copyright (C) 2018 Juan A. Rubio
This software is part of the Tizonia project <http://tizonia.org>

[Spotify] [Connecting] : 'myusername'.
[Spotify] [Album search] 'thriller'.
[Spotify] [Track] 'Wanna Be Startin' Somethin'' [Michael Jackson].
[Spotify] [Track] 'Baby Be Mine' [Michael Jackson].
[Spotify] [Track] 'The Girl Is Mine' [Michael Jackson].
[Spotify] [Track] 'Thriller' [Michael Jackson].
[Spotify] [Track] 'Beat It - Single Version' [Michael Jackson].
[Spotify] [Track] 'Billie Jean' [Michael Jackson].
[Spotify] [Track] 'Human Nature' [Michael Jackson].
[Spotify] [Track] 'P.Y.T. (Pretty Young Thing)' [Michael Jackson].
[Spotify] [Track] 'The Lady in My Life' [Michael Jackson].
[Spotify] [Track] 'Vincent Price Excerpt From "Thriller" Voice-Over Session - Thriller 25th Anniversary Voice-Over Session' [Vincent Price].
[Spotify] [Track] 'The Girl Is Mine 2008 with will.i.am - Thriller 25th Anniversary Remix' [Michael Jackson].
[Spotify] [Track] 'P.Y.T. (Pretty Young Thing) 2008 with will.i.am - Thriller 25th Anniversary Remix feat. willi.i.am' [Michael Jackson].
[Spotify] [Track] 'Wanna Be Startin' Somethin' 2008 with Akon - Thriller 25th Anniversary Remix' [Michael Jackson].
[Spotify] [Track] 'Beat It 2008 - Thriller 25th Anniversary Remix feat. Fergie' [Michael Jackson].
[Spotify] [Track] 'Billie Jean 2008 Kanye West Mix - Thriller 25th Anniversary Remix' [Michael Jackson].
[Spotify] [Track] 'For All Time' [Michael Jackson].
[Spotify] [Track] 'Voice-Over Intro Quincy Jones Interview #1 / Quincy Jones Interview #1' [Quincy Jones].
[Spotify] [Track] 'Someone in the Dark' [Michael Jackson].
[Spotify] [Track] 'Voice-Over Intro Quincy Jones Interview #2/Quincy Jones Interview #2 / Voice-Over Intro Billie Jean (Demo)' [Quincy Jones].
[Spotify] [Track] 'Billie Jean (Home Demo from 1981)' [Michael Jackson].
[Spotify] [Track] 'Quincy Jones Interview #3' [Quincy Jones].
[Spotify] [Track] 'Voice-Over Intro Rod Temperton Interview #1 / Rod Temperton Interview #1' [Rod Temperton].
[Spotify] [Track] 'Quincy Jones Interview #4' [Quincy Jones].
[Spotify] [Track] 'Voice-Over Intro / Voice-Over Session from Thriller' [Michael Jackson].
[Spotify] [Track] 'Voice-Over Intro Rod Temperton Interview #2 / Rod Temperton Interview #2' [Rod Temperton].
[Spotify] [Track] 'Quincy Jones Interview #5' [Quincy Jones].
[Spotify] [Track] 'Carousel' [Michael Jackson].
[Spotify] [Track] 'Voice-Over Intro Quincy Jones Interview #6 / Quincy Jones Interview #6' [Quincy Jones].
[Spotify] [Track] 'Billie Jean - Underground Mix' [Michael Jackson].
[Spotify] [Track] 'Thriller - Instrumental' [Michael Jackson].
[Spotify] [Tracks in queue] '30'.
[Spotify] : Spotify cache location: '/home/ra/snap/tizonia/common/tizonia-ra-spotify-myusername'
[Spotify] : 08:17:50.163 I [ap:1752] Connecting to AP ap.spotify.com:4070
[Spotify] : 08:17:50.165 I [user_cache:135] UserCache::initiateGetUsers() will query for 1 users
[Spotify] : 'myusername' logged in.
[Spotify] : 08:17:50.168 I [offline-mgr:2084] Storage has been cleaned
[Spotify] : 08:17:50.205 I [ap:1226] Connected to AP: 104.199.65.195:4070
[Spotify] [Connected] : 'thriller'.

tizonia exiting (OMX_ErrorInsufficientResources).

 [OMX.Aratelia.audio_renderer.pulseaudio.pcm:port:0]
 [OMX_ErrorInsufficientResources]

/snap/tizonia/193/bin/tizonia-snap-wrapper.sh: line 22:  1968 Segmentation fault      (core dumped) $SNAP/usr/bin/tizonia "$@"
```

Version
```
tizonia 0.15.0. Copyright (C) 2018 Juan A. Rubio
```

Debug
```
Debug Info:
	    * [Boost 1_58]
	    * [TagLib 1.9.1]
	    * [MediaInfoLib - v0.7.82]
```

OS
```
4.17.2-1-ARCH
```

**Comment by tizonia** [comment]
Hi!, thanks for taking the time to report this issue.

```
[OMX.Aratelia.audio_renderer.pulseaudio.pcm:port:0]
[OMX_ErrorInsufficientResources]
```

The error is reported by the PulseAudio renderer component. Do you have a working PulseAudio installation in your system? Also, has Tizonia ever worked for you on that system, or is it something that has just started to occur recently?

**Comment by RidaAyed** [comment]
Thanks

> Also, has Tizonia ever worked for you on that system, or is it something that has just started to occur recently?

No, it never worked

> Do you have a working PulseAudio ..?
```
$ pulseaudio --version
pulseaudio 12.0
```

**Comment by tizonia** [comment]
Thanks. 

Would you be able to upload Tizonia's debug logs?

See instructions here:
- https://github.com/tizonia/tizonia-snap/issues/3#issuecomment-398204642

Make sure you remove from the log things like your Spotify email and password before uploading.

Also, the sandbox denials info would be useful to see. Could you please run these commands (see https://docs.snapcraft.io/build-snaps/debugging) before running Tizonia (on a separate command window):
```
$ sudo snap install snappy-debug
$ sudo /snap/bin/snappy-debug.security scanlog tizonia
```

**Comment by tizonia** [comment]
@RidaAyed 

I think I may know what's going on. I believe you are running some 'non-mainstream' X desktop environment. 

Have you tried running Tizonia with Xfce or Gnome?. I've just tried running Tizonia with Xfce on my up-to-date Manjaro installation and it works perfectly fine. But I'm not a Xfce or Gnome user myself, and I normally run Awesome VM. It does not work there. What happens is that Tizonia (when installed as a snap) has problems with PulseAudio in Awesome VM and probably others.

This is a 'snapd' bug and has been open for a while, but still not resolved. I raised it myself, but somehow I keep forgetting that that issue still exists. Here is the bug:
https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1689539

In summary: 
- Tizonia (snap) should work with WMs like Xfce or Gnome. 
- If you want to run something like Awesome VM (and possibly others), you would need to install one of the AUR pacakges. But the AUR packages are currently dependent on libspotify, which has been removed from the AUR. See #499 .

Hopefully, I can update the AUR package to disable libspotify, but until then, Tizonia on Xfce or Gnome works fine.

**Comment by tizonia** [comment]
@RidaAyed 

I'm closing this one based on my previous comment. Please feel free to re-open if necessary.

**Comment by henioStraszny** [comment]
So I was trying to run it using WSL, and I've received similar error. Perhaps it was stupid idea that it could work but please let me know if there could be technical possibility for it to work on Windows Subsystem for Linux.

**Comment by MrBananane** [comment]
I am trying to run the dockerized version on MacOS and I have the exact same problem, even after installing and launching the pulseaudio service with Homebrew.

**Comment by mnymic** [comment]
Definitely, a DE-related issue.
For me, on Ubuntu 18.04 it works in Openbox but throws  OMX error in awesome.
